### DISCUSSION TITLE
How to disable Localization on build

### DISCUSSION THREAD

**Discussion by ns8482e** [OrchardCMS/OrchardCore#13662] [Q&A]
Is there a way to disable Localization copy on build for PO files?

I am trying to build a docker Image for web application with Core CMS Target - with PO files Image size is getting larger.

Currently  the audience for my application is only US English - and to reduce image size  - I really don't need to include localized PO files.

**Comment by hishamco** [comment]
I'm always disabling the MSBuild task, especially when I'm working from the source. IMHO we should not add all the translations by default, but if we do this changing the culture in the setup page will become useless

**Comment by jtkech** [comment]
One solution is to create a custom `OC.Application.Cms.Targets` that only reference the translation packages you need.

**Reply by ns8482e** [reply]
@jtkech you mean to  allow defining property for example `<ExcludeLocalization>true</ExcludeLocalization>`  or `<IncludeLocatizaion>fr-ca</IncludeLocatizaion>` in web project?

**Reply by jtkech** [reply]
Yes we could but I meant to define and reference your own project that only reference the translation packages that you need, this in place of referencing all translations as `OC.Application.Cms.Targets` does.

    <ItemGroup>
      <PackageReference Include="OrchardCore.Translations.All" />
    </ItemGroup>

Another way is to add in your web application

    <ItemGroup>
      <PackageTranslationFiles Remove="@(PackageTranslationFiles)" />
    </ItemGroup>

**Comment by willnationsdev** [comment]
For anyone who is still struggling with this..I attempted jtkech's solution (RIP), and it did not affect the generation of the Localization folder or the associated `.po` files when publishing; however, after peeking at the source code, it seems that they use a `DefaultItemExcludes` property to exclude items from the copying & publishing build targets. So you can effectively skip all the translation files by just adding them to the default excludes like so...

```msbuild
<PropertyGroup>
  <!-- other stuff -->
  
  <!-- Add the localization folder *itself* to the list of excluded items -->
  <DefaultItemExcludes>@(DefaultItemExcludes);Localization\**</DefaultItemExcludes>
</PropertyGroup>

<!-- If the Localization folder has been added to an ItemGroup, you may want to remove it. -->
<ItemGroup>
  <Folder Include="Localization\" /> <!-- <<<<< Remove this line -->
</ItemGroup>
```

If you do all this, then you will still see log messages during build about creating po files & copying/publishing them, but no actual work will be done as far as I can tell.

Edit: @hishamco You mentioned that you just directly disable the task itself. It might be helpful if you shared an example of how to do that since that may be a simpler solution than what I've detailed above.

Edit 2: Previously also hinted at doing `<SatelliteResourceLanguages>en-US;</SatelliteResourceLanguages>` since that stopped the language-specific .dll's from ending up in the publish output; however I then double-checked my web app and found that it started having 502 errors on startup after that change, so nevermind.

**Reply by hishamco** [reply]
Hope to look into that in this week
